src/main/java/mate/jdbc/dao/ManufacturerDao.java

```diff
@@ -0,0 +1,125 @@
+package mate.jdbc.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.exceptions.DaoException;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.models.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class ManufacturerDao implements mate.jdbc.dao.Dao<Manufacturer> {
+    private Manufacturer manufacturer = new Manufacturer();
+
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        String insertQuery = "INSERT INTO manufacturers(name, country) "
+                + "VALUES(?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement insertStatement = connection.prepareStatement(
+                        insertQuery,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            insertStatement.setString(1, manufacturer.getName());
+            insertStatement.setString(2, manufacturer.getCountry());
+            insertStatement.executeUpdate();
+            ResultSet resultSet = insertStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                manufacturer.setId(resultSet.getLong(1));
+            }
+        } catch (SQLException throwables) {
+            throw new DaoException("Can't insert manufacturer to db", throwables);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        String selectQuery = "SELECT * FROM manufacturers WHERE id = ? && is_deleted = false;";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement getQuery = connection.prepareStatement(selectQuery,
+                    Statement.RETURN_GENERATED_KEYS);
+            getQuery.setLong(1, id);
+            getQuery.executeQuery();
+            ResultSet resultSet = getQuery.getGeneratedKeys();
+            return Optional.ofNullable(parseResultSet(resultSet));
+        } catch (SQLException throwables) {
+            throw new DaoException("Can't get all formats from db", throwables);
```

Comment:
потрібно скрізь додати уточнення до тексту помилки
```suggestion
            throw new DaoException("Can't get manufacturer from db by id " + id, throwables);
```